src/batch/src/executor/join/lookup_join.rs

```diff
@@ -226,94 +228,128 @@ impl<P: 'static + ProbeSideSourceBuilder> LookupJoinExecutor<P> {
             None.into(),
         ));
 
-        while let Some(data_chunk) = build_side_stream.next().await {
-            let data_chunk = data_chunk?;
+        while let Some(build_chunk) = build_side_stream.next().await {
+            let build_chunk = build_chunk?.compact()?;
+
+            let mut min_heap = self.build_min_heap(&build_chunk)?;
+
+            while min_heap.peek().is_some() {
+                let mut row_idxs = vec![];
+                let mut prev_keys: Vec<ScalarImpl> = vec![];
```

Comment:
```suggestion
                let mut prev_key: Vec<ScalarImpl> = vec![];
```

Reply:
BTW I think `prev_key` should be named `curr_key` and `curr_key` should be `next_key` 🤔